File: frontend/src/types/distributionWeights.ts
export type DistributionWeightItem = {
  category: string;
  weight: number;
  defaultWeight: number;
  active: boolean;
  updatedAt: string;
  updatedBy: string;
};

export type DistributionWeightConfiguration = {
  items: DistributionWeightItem[];
  updatedAt: string | null;
  updatedBy: string | null;
};

export type DistributionWeightUpdateItem = Pick<DistributionWeightItem, "category" | "weight" | "active">;

File: backend/app/schemas/distribution_weights.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field, StrictInt


class DistributionWeightItem(BaseModel):
    category: str
    weight: StrictInt = Field(ge=1, le=5)
    active: bool


class DistributionWeightUpdateRequest(BaseModel):
    items: list[DistributionWeightItem]


class DistributionWeightResponseItem(DistributionWeightItem):
    defaultWeight: int
    updatedAt: datetime
    updatedBy: str


class DistributionWeightConfigurationResponse(BaseModel):
    items: list[DistributionWeightResponseItem]
    updatedAt: datetime | None = None
    updatedBy: str | None = None

